libcxx/include/typeinfo

```diff
@@ -57,365 +57,13 @@ public:
 */
 
```

Comment:
Let's do the header split i it's own PR instead.

Reply:
https://github.com/llvm/llvm-project/pull/68812

Reply:
We decided to drop backwards compatibility include (typeinfo -> exception) instead